src/core/graph.js

```diff
@@ -7,6 +7,80 @@ import {makeAddressModule, type AddressModule} from "./address";
 import {toCompat, fromCompat, type Compatible} from "../util/compat";
 import * as NullUtil from "../util/null";
 
+/**
+ * This module contains the Graph, which is one of the most fundamental pieces
+ * of SourceCred. SourceCred uses this graph to model all of the contributions
+ * that make up a project, and the relationships between those contributions.
+ *
+ * The Graph serves a simple function: it keeps track of which Nodes exist, and
+ * what Edges join those nodes to each other. Nodes and Edges are both identified
+ * by Addresses; specifically, `NodeAddressT`s and `EdgeAddressT`s.
+ *
+ * In both cases, addresses are modeled as arrays of strings. For example,
+ * you might want to give an address to your favorite node. You can do so as
+ * follows:
+ *
+ * const myAddress: NodeAddressT = NodeAddress.fromParts(["my", "favorite"])
+ *
+ * Edge Addresses are quite similar, except you use the EdgeAddress module.
+ *
+ * We model addresses as arrays of strings so that plugins can apply
+ * hierarchical namespacing for the address. In general, for any address, the
+ * first piece should be the name of the organization that owns the plugin, and
+ * the second piece should be the name of the plugin. Pieces thereafter are
+ * namespaced by the plugin's internal logic. For example, SourceCred has a
+ * Git plugin, and that plugin produces addresses like
+ * ["sourcecred", "git", "commit", "9cba0e9e212a287ce26e8d7c2d273e1025c9f9bf"].
+ *
+ * This enables "prefix matching" for finding only certain types of nodes. For
+ * example, if we wanted to find every Git commit in the graph, we
+ * could use the following code:
+ *
+ * const commitPrefix = NodeAddress.fromParts(["sourcecred", "git", "commit"])
+ * const commitNodes = `graph.nodes({prefix: commitPrefix})`
+ *
+ * Under the hood, addresses are represented as strings. Basically, we
+ * concatenate the pieces of the address together with the null seperator, and
+ * prepend a nonce ("E" or "N") to distinguish node and edge addresses. So the
+ * address for the node with pieces ["my", "favorite"] would be
+ * "N\0my\0favorite". Representing addreses as strings means that we can use
+ * them as keys for maps. To see the implementation, and the full address model
+ * API, check out src/core/address.js.
+ *
+ * In the case of a node, the address is all the graph knows about. If you have
+ * any other metadata you want to store about the node—like its commit message
+ * or its favorite color—you can store that in another database. The Graph is
+ * just a graph, not a key value store.
+ *
+ * Edges need a little more data, since the graph needs to know what nodes the
+ * edge connects. Each edge is an object with `src` and `dst` fields which are
```

Comment:
What is the significance of the `src` and `dst` fields?

Reply:
The "source" and "destination" of the edge. Re-wrote the paragraph to clarify :+1: